langtest/datahandler/datasource.py

```diff
@@ -99,7 +98,11 @@ def __init__(self, file_path: str, task: str, **kwargs) -> None:
             file_path (str): Path to the dataset.
             task (str): Task to be evaluated.
         """
-        self._file_path = file_path
+        self._custom_label = file_path
+        if isinstance(self._custom_label, dict):
+            self._file_path = file_path["name"]
+        else:
+            self._file_path = file_path
```

Comment:
Don't get what is this `_custom_label` attribute

Reply:
`_custom_label`  conatin

```
{"name": r"data\imdb.csv",
                        "feature_column": "text",    
                        "target_column": 'label',     
                       }```

Reply:
Well then `file_path` is not a string anymore and doesn't refer to the file location

Reply:
`file_path` contains only the path i think , 
also based on the path extension we are calling the `load_data` method of respective class